### PR TITLE
Add capacity reservation group workflow to azure-compute skill

### PR DESCRIPTION
**Pull request by andersone181**
## Description

* Added the capacity reservation group workflow to the `azure-compute` skill
    * Assists with finding existing capacity reservations, associating/disassociating reservations with VMs

Reference files:
* Added `capacity-reservation-overview.md` for general guidance and specific reference for finding compatible capacity reservations
* Added `association-disassociation.md` for guidance on detaching a CRG from a VM

## Checklist

- [x] Tests pass locally (`cd tests && npm test`)
- [x] **If modifying skill descriptions:** verified routing correctness with integration tests (`npm run test:skills:integration -- <skill>`)
- [x] **If modifying skill `USE FOR` / `DO NOT USE FOR` / `PREFER OVER` clauses:** confirmed no routing regressions for competing skills
- [x] Version bumped in skill frontmatter (if skill files changed)

passing_compute_tests_latest

### CODE REVIEW

**Comment by andersone181** [comment]
@microsoft-github-policy-service agree company="Microsoft"

**Review by alex-thompson — DISMISSED** [review]

File: plugin/skills/azure-compute/workflows/capacity-reservation/capacity-reservation.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  -n <crg-name> \
+  -l <region> \
+  --zones <zone>           # optional — omit if non-zonal
+
+# Create the reservation
+az capacity reservation create \
+  -g <resource-group> \
+  -c <crg-name> \
+  -n <reservation-name> \
+  --sku <vm-size> \
```
[the remark this answers is not in the document]
Comment (andersone181) [reply plugin/skills/azure-compute/workflows/capacity-reservation/capacity-reservation.md:77]: Zones is a supported property on CRG, it needs to be specified to create a zonal CRG instead of a regional one. I updated this to make it clear that zones/zone are optional on group/reservation create but are required to create zonal reservations.

File: plugin/skills/azure-compute/workflows/capacity-reservation/references/capacity-reservation-overview.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+| List CRGs                   | `az capacity reservation group list`                                                                        |
+| Show CRG                    | `az capacity reservation group show -g <rg> -n <crg> --instance-view`                                       |
+| Delete CRG                  | `az capacity reservation group delete -g <rg> -n <crg>`                                                     |
+| List reservations           | `az capacity reservation list -g <rg> --capacity-reservation-group <crg>`                                   |
+| Update reservation quantity | `az capacity reservation update -g <rg> --capacity-reservation-group <crg> -n <res> --capacity <new-count>` |
+| Delete reservation          | `az capacity reservation delete -g <rg> --capacity-reservation-group <crg> -n <res>`                        |
+
+## Finding Valid CRGs for a VM
+
+To associate a VM with a CRG, the CRG must contain a capacity reservation that matches the VM's **size**, **region**, and **zone** (if zonal). Since CRGs are listed per-resource-group via CLI, discovering the right one across multiple resource groups requires querying at the subscription level.
```
[the remark this answers is not in the document]
Comment (andersone181) [reply plugin/skills/azure-compute/workflows/capacity-reservation/references/capacity-reservation-overview.md]: Reworded to make the guidance consistent with the CLI command recommendation.

File: plugin/skills/azure-compute/SKILL.md [hunk 3]
```diff
@@ -1,10 +1,10 @@
 ---
 name: azure-compute
-description: "Azure VM and VMSS router for recommendations, pricing, autoscale, orchestration, and connectivity troubleshooting. WHEN: Azure VM, VMSS, scale set, recommend, compare, server, website, burstable, lightweight, VM family, workload, GPU, learning, simulation, dev/test, backend, autoscale, load balancer, Flexible orchestration, Uniform orchestration, cost estimate, connect, refused, Linux, black screen, reset password, reach VM, port 3389, NSG, troubleshoot."
+description: "Azure VM and VMSS router for recommendations, pricing, autoscale, orchestration, connectivity troubleshooting, and capacity reservations. WHEN: Azure VM, VMSS, scale set, recommend, compare, server, website, burstable, lightweight, VM family, workload, GPU, learning, simulation, dev/test, backend, autoscale, load balancer, Flexible orchestration, Uniform orchestration, cost estimate, connect, refused, Linux, black screen, reset password, reach VM, port 3389, NSG, troubleshoot, capacity reservation, CRG, reserve VMs, guarantee capacity, pre-provision capacity, associate VM with CRG, disassociate VM from CRG."
```
[the remark this answers is not in the document]
Comment (andersone181) [reply plugin/skills/azure-compute/SKILL.md]: Removed with/from and reran the integration tests

File: plugin/skills/azure-compute/workflows/capacity-reservation/references/capacity-reservation-overview.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+### Option 2: CLI enumeration
+
+If ARG is unavailable, list CRGs per resource group and inspect their reservations:
+
+```bash
+# List all CRGs
+az capacity reservation group list -o table
+
+# List reservations within a CRG and check for matching size/capacity
+az capacity reservation list \
```
Comment (jongio) [inline plugin/skills/azure-compute/workflows/capacity-reservation/references/capacity-reservation-overview.md:81]: `az capacity reservation list` here is missing the required `-g <rg>` parameter. Without it, the command fails unless the user has a default resource group configured (most won't).

Should be:
```bash
az capacity reservation list \
  -g <rg> \
  --capacity-reservation-group <crg-name> \
  --query "[?sku.name=='<vm-size>'].{name:name, size:sku.name, capacity:sku.capacity, zones:zones}" \
  -o table
```

File: plugin/skills/azure-compute/workflows/capacity-reservation/references/association-disassociation.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+└── capacityReservationGroup.id = <CRG resource ID>
+    └── Azure auto-matches to a reservation with the right VM size + zone
+```
+
+### Associating VMs
+
+When creating or updating a VM, set the `capacityReservationGroup` property:
+
+```bash
+az vm create \
```
Comment (jongio) [inline plugin/skills/azure-compute/workflows/capacity-reservation/references/association-disassociation.md:23]: The text says "creating or updating" but only `az vm create` is shown. For existing VMs, users need `az vm update` - and the VM must be deallocated first. This is a common scenario ("I already have a VM, how do I add it to a CRG?").

Consider adding an "Existing VM" example:
```bash
# Existing VM - deallocate first, then associate:
az vm deallocate -g <rg> -n <vm-name>
az vm update -g <rg> -n <vm-name> --capacity-reservation-group <crg-id>
az vm start -g <rg> -n <vm-name>
```

File: tests/azure-compute/triggers.test.ts [hunk 6]
```diff
@@ -234,7 +256,7 @@ describe(`${SKILL_NAME} - Trigger Tests`, () => {
       "Configure my PostgreSQL database", // Different service, no azure keyword
       "How do I write a Python web scraper?", // Unrelated to Azure
       "Set up a Kubernetes cluster with Helm", // AKS, not VMs
-      "Create a serverless function with AWS Lambda", // Wrong cloud provider
+      "Create a serverless function on AWS", // Wrong cloud provider
```
Comment (jongio) [inline tests/azure-compute/triggers.test.ts:259]: Nit: the new `capacity` keyword is broad enough to match quota or AI-capacity prompts. Consider adding a negative trigger test like `Check my Azure OpenAI capacity limits` to guard against cross-skill false routing.

**Review by jongio — COMMENTED** [review]
Solid addition - the workflow structure, progressive disclosure, and test coverage all follow the existing patterns well. Three items to address:

1. Missing `-g` param in CLI enumeration example (capacity-reservation-overview.md)
2. Association section only covers `az vm create` but not `az vm update` for existing VMs (association-disassociation.md)
3. Consider a negative trigger test for the broad `capacity` keyword

Also worth noting: the two Copilot bot comments about `--capacity-reservation-group None` being invalid are incorrect - that IS the documented Azure CLI syntax for clearing a CRG association. You can safely dismiss those.

File: plugin/skills/azure-compute/workflows/capacity-reservation/references/capacity-reservation-overview.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+            capacity = coalesce(sku.capacity, 0),
+            associationCount = coalesce(array_length(properties.virtualMachinesAssociated), 0),
+            location
+" --query "data[]" -o table
+```
+
+The `crgId` in the output is the parent Capacity Reservation Group resource ID — this is the value to use when associating a VM or VMSS.
+
+To further narrow results for zonal VMs, add a zone filter:
+
```
[the remark this answers is not in the document]
Comment (andersone181) [reply plugin/skills/azure-compute/workflows/capacity-reservation/references/capacity-reservation-overview.md:67]: This seems to be a hallucination, the example query uses a single `|`

File: plugin/skills/azure-compute/workflows/capacity-reservation/references/association-disassociation.md [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+There are three ways to disassociate. The commands below use `az vm` — for VMSS, substitute `az vmss` and add `az vmss update-instances --instance-ids "*"` as a final step when using a **Manual** upgrade policy.
+
+### Option 1: Deallocate, then remove association
+
+Best when the VM/VMSS can tolerate downtime.
+
+```bash
+az vm deallocate -g <rg> -n <vm-name>
+az vm update -g <rg> -n <vm-name> --capacity-reservation-group None
+az vm start -g <rg> -n <vm-name>   # optional
```
[the remark this answers is not in the document]
Comment (andersone181) [reply plugin/skills/azure-compute/workflows/capacity-reservation/references/association-disassociation.md:76]: This is the recommended detach method from the [documentation](https://learn.microsoft.com/en-us/azure/virtual-machines/capacity-reservation-remove-vm?tabs=cli1%2Ccli2#deallocate-the-virtual-machine)

File: plugin/skills/azure-compute/workflows/capacity-reservation/references/association-disassociation.md [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+
+### Option 2: Set reserved quantity to zero, then remove association
+
+Best when the VM/VMSS cannot be deallocated and the reservation is no longer needed.
+
+```bash
+az capacity reservation update \
+  -g <rg> --capacity-reservation-group <crg> \
+  -n <reservation-name> --capacity 0
+az vm update -g <rg> -n <vm-name> --capacity-reservation-group None
```
[the remark this answers is not in the document]
Comment (andersone181) [reply plugin/skills/azure-compute/workflows/capacity-reservation/references/association-disassociation.md:87]: See other comment, this is the method used in the official CRG docs

File: tests/azure-compute/triggers.test.ts (hunk 6 above)
Comment (andersone181) [reply tests/azure-compute/triggers.test.ts:259]: When I use this prompt in particular, "Check my Azure OpenAI capacity limits", the trigger test does incorrectly route to the capacity reservation group skill, because of the "azure" and "capacity" keywords in the prompt.

Am I correct in my understanding though that the LLM is using more context than the word matching done by the trigger test? So the actual behavior might be okay, even if we can't create a negative trigger test?

I tried inputting the prompt into Copilot, and it did not call into the `azure-compute` skill when I did, it chose `azure-quotas` instead. Which seems to indicate to me that the routing in practice is okay.

File: tests/azure-compute/unit.test.ts [hunk 10]
```diff
@@ -363,6 +370,20 @@ describe(`${SKILL_NAME} - Unit Tests`, () => {
...
+        ),
+        "utf-8"
+      );
+      crAssociationContent = await fs.readFile(
+        path.join(
+          SKILLS_PATH,
+          "azure-compute/workflows/capacity-reservation/references/association-disassociation.md"
+        ),
+        "utf-8"
+      );
```
[the remark this answers is not in the document]
Comment (andersone181) [reply tests/azure-compute/unit.test.ts:397]: Added a constant with the path to address this

File: plugin/skills/azure-compute/workflows/capacity-reservation/references/capacity-reservation-overview.md (hunk 4 above)
Comment (andersone181) [reply plugin/skills/azure-compute/workflows/capacity-reservation/references/capacity-reservation-overview.md:81]: Updated, added the rg parameter

File: plugin/skills/azure-compute/workflows/capacity-reservation/references/association-disassociation.md (hunk 5 above)
Comment (andersone181) [reply plugin/skills/azure-compute/workflows/capacity-reservation/references/association-disassociation.md:23]: I reworked this, added an explicit update example for VM and a note for VMSS. Also added notes on when it is necessary to deallocate the VM (regional only, no need to deallocate to attach a zonal CRG)

File: plugin/skills/azure-compute/workflows/capacity-reservation/references/capacity-reservation-overview.md [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+To associate a VM with a CRG, the CRG must contain a capacity reservation that matches the VM's **size**, **region**, and **zone** (if zonal). While `az capacity reservation group list` can enumerate CRGs at the subscription level, filtering down to matching reservations across many groups is inefficient. Azure Resource Graph is recommended for cross-resource-group discovery.
+
+### Option 1: Azure Resource Graph (recommended)
+
+ARG can query all capacity reservations across resource groups in a single call, filtering by location, VM size, and zone. This is the most efficient approach.
+
+> ⚠️ **Prerequisite:** `az extension add --name resource-graph`
+
+You must collapse this query to a **single line** before running it:
```
[the remark this answers is not in the document]
Comment (andersone181) [reply plugin/skills/azure-compute/workflows/capacity-reservation/references/capacity-reservation-overview.md:44]: I see no reason not to instruct Copilot to always collapse the query.

File: tests/azure-compute/integration.test.ts [hunk 12]
```diff
@@ -158,5 +159,33 @@ describeIntegration(`${SKILL_NAME}_ - Integration Tests`, () => {
...
       });
     });
+
+    test("routes capacity reservation creation prompt to capacity-reservation", async () => {
+      await withTestResult(async ({ setSkillInvocationRate }) => {
+        const result = await expectPromptToInvokeWorkflow(
+          "I need to create a Capacity Reservation Group in Azure to guarantee Standard_D4s_v5 capacity in East US.",
+          CAPACITY_RESERVATION_WORKFLOW_PATH,
+        );
+        if (!result) return;
```
[the remark this answers is not in the document]
Comment (andersone181) [reply tests/azure-compute/integration.test.ts:169]: This follows the existing pattern

**Review by jongio — COMMENTED** [review]
Addressed my earlier feedback. One note: the open Copilot comment about `||` in tables on capacity-reservation-overview.md is a false positive - that file uses single `|` at row starts. Safe to resolve.

File: plugin/skills/azure-compute/workflows/capacity-reservation/references/capacity-reservation-overview.md [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+
+## Constraints
+
+| Constraint                     | Detail                                                                                                     |
+|--------------------------------|------------------------------------------------------------------------------------------------------------|
+| **Region-scoped**              | A CRG and all its reservations must be in the same Azure region                                            |
+| **Zone-specific**              | Each reservation targets a specific Availability Zone (or is non-zonal)                                    |
+| **Subscription-scoped**        | A CRG lives in a single subscription but can be shared with other subscriptions via the `sharing` property |
+| **VM size per reservation**    | Each capacity reservation covers exactly one VM size                                                       |
+| **Billing starts immediately** | You are charged for reserved capacity whether or not VMs are running against it                            |
```
[the remark this answers is not in the document]
Comment (jongio) [reply plugin/skills/azure-compute/workflows/capacity-reservation/references/capacity-reservation-overview.md:17]: False positive - capacity-reservation-overview.md uses single `|` at row starts, not `||`. Verified against the current file on this PR.

File: tests/azure-compute/triggers.test.ts (hunk 6 above)
Comment (andersone181) [reply tests/azure-compute/triggers.test.ts:259]: Spoke with @alex-thompson offline, I think this will be fine as is.

**Review by alex-thompson — APPROVED** [review]
